### PR TITLE
Add a doctest example for AttentionWrapper

### PR DESCRIPTION
**Pull request by guillaumekln**
# Description

This documents a typical usage of the wrapper.

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

### CODE REVIEW

File: tensorflow_addons/seq2seq/attention_wrapper.py [hunk 1]
```diff
@@ -1562,7 +1562,31 @@ def _compute_attention(
...
+
+    >>> batch_size = 4
+    >>> max_time = 7
+    >>> hidden_size = 32
+    >>>
+    >>> memory = tf.random.uniform([batch_size, max_time, hidden_size])
+    >>> memory_sequence_length = tf.fill([batch_size], max_time)
+    >>>
+    >>> attention_mechanism = tfa.seq2seq.LuongAttention(hidden_size)
+    >>> attention_mechanism.setup_memory(memory, memory_sequence_length)
```
Comment (qlzh727) [inline tensorflow_addons/seq2seq/attention_wrapper.py:1577]: Do we suggest user to call setup_memory() explicitly, or just pass the memory related param in the __init__()?

File: tensorflow_addons/seq2seq/attention_wrapper.py [hunk 2]
```diff
@@ -1562,7 +1562,31 @@ def _compute_attention(
...
+
+    >>> cell = tf.keras.layers.LSTMCell(hidden_size)
+    >>> cell = tfa.seq2seq.AttentionWrapper(
+    ...     cell, attention_mechanism, attention_layer_size=hidden_size)
+    >>>
+    >>> inputs = tf.random.uniform([batch_size, hidden_size])
+    >>> state = cell.get_initial_state(inputs)
+    >>>
+    >>> outputs, state = cell(inputs, state)
```
Comment (qlzh727) [inline tensorflow_addons/seq2seq/attention_wrapper.py:1586]: I think the state can be skipped since the cell will be default call cell.get_initial_state() if the state is not passed in the call().

File: tensorflow_addons/seq2seq/attention_wrapper.py (hunk 1 above)
Comment (guillaumekln) [reply tensorflow_addons/seq2seq/attention_wrapper.py:1577]: Which one do you think is a better example? I used `setup_memory` because it sets the path on how to build the model first and run it later.

In any case, we add a comment indicating that there is another way to set the memory.

File: tensorflow_addons/seq2seq/attention_wrapper.py (hunk 2 above)
Comment (guillaumekln) [reply tensorflow_addons/seq2seq/attention_wrapper.py:1586]: The state argument [is not optional](https://github.com/tensorflow/addons/blob/v0.11.2/tensorflow_addons/seq2seq/attention_wrapper.py#L1949) for this cell, so I think we should keep it.

Comment (qlzh727) [reply tensorflow_addons/seq2seq/attention_wrapper.py:1586]: oh, you are right. The state is optional for lstm_cell, but not for the wrapper.

**Review by qlzh727 — APPROVED** [review]
